Kernel build breaks with 

| In file included from fs/btrfs/ctree.c:21:0:
| fs/btrfs/ctree.h:1003:17: error: field 'super_kobj' has incomplete type
| fs/btrfs/ctree.h:1074:17: error: field 'root_kobj' has incomplete type
| make[2]: *** [fs/btrfs/ctree.o] Error 1
| make[1]: *** [fs/btrfs] Error 2

Fix found here: https://lkml.org/lkml/2011/1/12/76

--- git/fs/btrfs/ctree.h	2011-11-23 19:20:42.206976619 +0100
+++ git-neu/fs/btrfs/ctree.h	2011-11-23 19:20:27.628331541 +0100
@@ -25,6 +25,7 @@
 #include <linux/fs.h>
 #include <linux/completion.h>
 #include <linux/backing-dev.h>
+#include <linux/kobject.h>
 #include <linux/wait.h>
 #include <linux/slab.h>
 #include <asm/kmap_types.h>
